### PR TITLE
fix: Do not check for key existence after first time (resolves api limits and failing secrets)

### LINKED ISSUE #52: AAD too many requests
We got 429 Too Many Requests error from AAD after we restarted API Server. It lasted for quite a while and caused API Server outage since all secrets were unable to be read.

After reading the code, I find that each time KMS makes a request to KeyVault, it calls `getKey` first. Inside it creates a new KeyVaultClient along with a new AAD token, which means a new call to AAD later. I think that's unnecessary because you only need to create a new key when it doesn't exist. And once it's created, its version is persisted and never changes. We don't have to check it in each encryption or decryption calls.

I'm wondering if we can move the `getKey` call to KMS starting time and save the KeyVaultClient in `KeyManagementServerService` for later reuse. In this case, AAD token is also able to be reused. If you can confirm this, I'm happy to submit a PR :)

@ritazh

### PR DESCRIPTION
**Pull request by zachomedia**
**Reason for Change**:

We hit the issue identified in #52 after upgrading our Kubernetes clusters. controller-manager wouldn’t start because it kept erroring trying to sync secrets. We looked into the kube-azure-kms pod and saw a bunch of auth errors and some searching led us to that issue.

We decided to try the fix that was suggested and it seems to have worked. We're not sure if there are any impacts from this change though.

**Issue Fixed**:

Fixes #52 

**Notes for Reviewers**:

We've been using this for about 1 month internally, with no issues. However, as mentioned, I'm not sure if there are any unintended consequences of this change.

I had to update dependencies to get it to build (so that's why there are additional changes). I can try to PR without them if you prefer.

### CODE REVIEW

**Comment by aramase** [comment]
The unit tests pipeline should be fixed with this PR https://github.com/Azure/kubernetes-kms/pull/67

File: Makefile [hunk 1]
```diff
@@ -34,9 +34,9 @@ push: build-image
 
 .PHONY: clean deps unit-test integration-test
 
-deps: setup
-	@echo "Ensuring Dependencies..."
-	$Q go env
+# deps: setup
```
Comment (aramase) [inline Makefile:37]: Could you remove this target from the Makefile as it's no longer required?

File: server.go [hunk 2]
```diff
@@ -94,16 +94,28 @@ func New(pathToUnixSocketFile string, configFilePath string) (*KeyManagementServ
 	return keyManagementServiceServer, nil
 }
 
+var kvClient *kv.ManagementClient = nil
```
Comment (aramase) [inline server.go:97]: Could you group these variables and move this to the top of the package? The default pointer value will be nil, so we don't need to initialize it again. 

```
var (
  kvClient *kv.ManagementClient
  vaultUrl *string
)
```

**Review by aramase — COMMENTED** [review]
Thank you for the PR. Added few minor comments.

File: Makefile (hunk 1 above)
Comment (zachomedia) [reply Makefile]: Done!

File: server.go (hunk 2 above)
Comment (zachomedia) [reply server.go]: Done!

**Review by ritazh — APPROVED** [review]

**Review by aramase — APPROVED** [review]

**Pull request merged**